application/common/models/Method.php

```diff
@@ -131,7 +140,7 @@ public static function createAndSendVerification($userId, $type, $value)
                 throw new BadRequestHttpException($e->getMessage(), $e->getCode());
             }
         } else {
```

Comment:
Similar to my comment below, it may help readability (and possible future error conditions) to make this 
`} else {` 
an 
`} elseif ($type === self::TYPE_EMAIL) {`
and add a separate else to handle unknown types.